### Title: 09_NCBI_Genomes_Mapping_SetB ###
### Author: Ema H Graham ###
### For Questions Email: ema.graham@huskers.unl.edu ###

############################################
## -------------------------------------- ##
## ------------- Description ------------ ##
## -------------------------------------- ##
############################################

#Input Notes:
#Reference genome sequences (i.e., caudovirales.fasta,papilloma.fasta, etc.) were obtained from NCBI's nucleotide data and contain all (full genome) sequences that are available for genomes that are taxonomically classified as either Caudovirales, Papillomaviridae, Genomoviridae, and Baculoviridae.
#These Reference fasta files were accessed and downloaded on May 24th, 2021.
#Need QC trimmed and contamination removed reads generated in 02_Virome_Assembly.sh, located in Human_Remove_Trimmed/
  #Trimmed reads were used to prevent mapping of reference reads to host or phix contamination within the sequences
    #This was also done with just mapping to raw reads and the same results were observed (not reported)

#Output Notes:
#This pipeline will generate a read abundance tables for each reference type that can be used for input into Phyloseq in R (counts2_CAUDO.txt, PAPcounts.txt, counts2_genomovir.txt, counts2_baculo.txt)
#The generated read abundance tables were then opened on my desktop and saved as .csv files for easier import into R. 
#The next step of the pipeline is 10_Set_B.Rmd

#General Notes:
#This pipeline is designed to be run using the Holland Computing Center at the University of Nebraska. Some tool commands may differ depending on installation of the tool. Please refer to the listed Githubs for each tool used as mentioned in script for further information if issues arise 
#Some file locations may differ from yours so this needs to be changed accordingly. This script is designed to be run all in one folder
#For the steps indicated make sure to repeat the pipeline so that all non-negative control samples are run through the pipeline. Examples were given for one sample at each step (HV_001_01)


############################################
## -------------------------------------- ##
## - Build Caudovirales Reference Index - ##
## -------------------------------------- ##
############################################

mkdir Caudo/
cd Caudo/

#The data file caudovirales.fasta was obtained from NCBI and contains all NCBI nucleotide genomic sequences that are taxonomically classified as being Caudovirales
#Move caudovirales.fasta to this folder (i.e. Caudo/)

#The global alignment tool Bowtie2 v.2.3.5 was used for indexing and mapping
#Bowtie2 can be found at: http://bowtie-bio.sourceforge.net/bowtie2/index.shtml

bowtie2-build -f caudovirales.fasta CAUDOVIRALES_DB

############################################
## -------------------------------------- ##
## --- Caudovirales Reference Mapping --- ##
## -------------------------------------- ##
############################################

#Mapping was performed for each sample (not for negative controls) so repeat this step for each QC trimmed individual sample. Here we use the example of using sample HV_001_01

bowtie2 --end-to-end -x CAUDOVIRALES_DB -1 ../Human_Remove_Trimmed/HV_001_01_final_R1.fastq -2 ../Human_Remove_Trimmed/HV_001_01_final_R2.fastq | samtools view -F 4 -o HV_001_01.bam

#Samtools was used for subsiquent manipluation of the sam file output from bowtie2
#Samtools can be found at: https://github.com/samtools/samtools

samtools sort -O BAM -o HV_001_01.sorted.bam HV_001_01.bam 
samtools index HV_001_01.sorted.bam
samtools idxstats HV_001_01.sorted.bam > HV_001_01.sorted.bam.idxstats.txt

#################################################################
## ----------------------------------------------------------- ##
## - Caudovirales Reference Mapping to Abundance Table for R - ##
## ----------------------------------------------------------- ##
#################################################################

#Once all samples have been run through this pipeline you can convert the mapped abundance statistics into a read abundance table that can be used as an abundance table in R. 
#get_count_table.py python script avalible at: https://github.com/edamame-course/Metagenome

python2 get_count_table.py *.idxstats.txt > counts_CAUDO.txt
sed 's/.idxstats.txt//g' counts_CAUDO.txt > counts2_CAUDO.txt

###########################################################
## ----------------------------------------------------- ##
## ------------- Caudovirales Taxonomy File ------------ ##
## ----------------------------------------------------- ##
###########################################################

#The Caudovirales taxonomy file (CAUDO_tax.csv) was generated by taking all the accession numbers associated with the reference genomes (caudovirales.fasta) used and putting them in an excel file
#All accession numbers that did not have any mapped reads across all samples were removed from the file
#The remaining accession numbers (that had at least one mapped read in a sample) were annotated using their associated taxonomy information provided on NCBI

################################################
## ------------------------------------------ ##
## - Build Papillomaviridae Reference Index - ##
## ------------------------------------------ ##
################################################

cd ../
mkdir Papilloma/
cd Papilloma/

#The data file papillomaviridae.fasta was obtained from NCBI and contains all NCBI nucleotide genomic sequences that are taxonomically classified as being Papillomaviridae
#Move papillomaviridae.fasta to this folder (i.e. Papilloma/)

bowtie2-build -f papillomaviridae.fasta PAPILLOMA_DB

############################################
## -------------------------------------- ##
## - Papillomaviridae Reference Mapping - ##
## -------------------------------------- ##
############################################

#Mapping was performed for each sample (not for negative controls) so repeat this step for each individual sample. Here we use the example of using sample HV_001_01

bowtie2 --end-to-end -x PAPILLOMA_DB -1 ../Human_Remove_Trimmed/HV_001_01_final_R1.fastq -2 ../Human_Remove_Trimmed/HV_001_01_final_R2.fastq | samtools view -F 4 -o HV_001_01.bam

samtools sort -O BAM -o HV_001_01.sorted.bam HV_001_01.bam 
samtools index HV_001_01.sorted.bam
samtools idxstats HV_001_01.sorted.bam > HV_001_01.sorted.bam.idxstats.txt

#####################################################################
## --------------------------------------------------------------- ##
## - Papillomaviridae Reference Mapping to Abundance Table for R - ##
## --------------------------------------------------------------- ##
#####################################################################

#Once all samples have been run through this pipeline you can convert the mapped abundance statistics into a read abundance table that can be used as an abundance table in R. 

python2 get_count_table.py *.idxstats.txt > counts_PAP.txt
sed 's/.idxstats.txt//g' counts_PAP.txt > PAPcounts.txt

############################################
## -------------------------------------- ##
## --- Papillomaviridae Taxonomy File --- ##
## -------------------------------------- ##
############################################

#The Papillomaviridae taxonomy file (papilloma_taxa.csv) was generated by taking all the accession numbers associated with the reference genomes (papillomaviridae.fasta) used and putting them in an excel file
#The accession numbers were annotated using their associated taxonomy information provided on NCBI

#############################################
## --------------------------------------- ##
## - Build Genomoviridae Reference Index - ##
## --------------------------------------- ##
#############################################

cd ../
mkdir Genomo/
cd Genomo/

#The data file genomoviridae.fasta was obtained from NCBI and contains all NCBI nucleotide genomic sequences that are taxonomically classified as being Genomoviridae
#Move genomoviridae.fasta to this folder (ie Genomo/)

bowtie2-build -f genomoviridae.fasta GENOMO_DB

###########################################
## ------------------------------------- ##
## -- Genomoviridae Reference Mapping -- ##
## ------------------------------------- ##
###########################################

#Mapping was performed for each sample (not for negative controls) so repeat this step for each individual sample. Here we use the example of using sample HV_001_01

bowtie2 --end-to-end -x GENOMO_DB -1 ../Human_Remove_Trimmed/HV_001_01_final_R1.fastq -2 ../Human_Remove_Trimmed/HV_001_01_final_R2.fastq | samtools view -F 4 -o HV_001_01.bam

samtools sort -O BAM -o HV_001_01.sorted.bam HV_001_01.bam 
samtools index HV_001_01.sorted.bam
samtools idxstats HV_001_01.sorted.bam > HV_001_01.sorted.bam.idxstats.txt

##################################################################
## ------------------------------------------------------------ ##
## - Genomoviridae Reference Mapping to Abundance Table for R - ##
## ------------------------------------------------------------ ##
##################################################################

#Once all samples have been run through this pipeline you can convert the mapped abundance statistics into a read abundance table that can be used as an abundance table in R. 

python2 get_count_table.py *.idxstats.txt > counts_genomovir.txt
sed 's/.idxstats.txt//g' counts_genomovir.txt > counts2_genomovir.txt

#this was further edited in excel to remove all reference accession genomes that had zero mapped reads across all samples to make the file smaller in size (counts2_genomovir_edited.csv)


############################################
## -------------------------------------- ##
## ----- Genomoviridae Taxonomy File ---- ##
## -------------------------------------- ##
############################################

#The Genomoviridae taxonomy file (genomovir_taxa.csv) was generated by taking all the accession numbers associated with the reference genomes (genomoviridae.fasta) used and putting them in an excel file
#All accession numbers that did not have any mapped reads across all samples were removed from the file
#The remaining accession numbers (that had at least one mapped read in a sample) were annotated using their associated taxonomy information provided on NCBI

#############################################
## --------------------------------------- ##
## - Build Baculoviridae Reference Index - ##
## --------------------------------------- ##
#############################################

cd ../
mkdir Baculo/
cd Baculo/

#The data file baculoviridae.fasta was obtained from NCBI and contains all NCBI nucleotide genomic sequences that are taxonomically classified as being Baculoviridae
#Move baculoviridae.fasta to this folder (i.e. Baculo/)

bowtie2-build -f papillomaviridae.fasta BACULO_DB

#############################################
## --------------------------------------- ##
## --- Baculoviridae Reference Mapping --- ##
## --------------------------------------- ##
#############################################

#Mapping was performed for each sample (not for negative controls) so repeat this step for each individual sample. Here we use the example of using sample HV_001_01

bowtie2 --end-to-end -x BACULO_DB -1 ../Human_Remove_Trimmed/HV_001_01_final_R1.fastq -2 ../Human_Remove_Trimmed/HV_001_01_final_R2.fastq | samtools view -F 4 -o HV_001_01.bam

samtools sort -O BAM -o HV_001_01.sorted.bam HV_001_01.bam 
samtools index HV_001_01.sorted.bam
samtools idxstats HV_001_01.sorted.bam > HV_001_01.sorted.bam.idxstats.txt

##################################################################
## ------------------------------------------------------------ ##
## - Baculoviridae Reference Mapping to Abundance Table for R - ##
## ------------------------------------------------------------ ##
##################################################################

#Once all samples have been run through this pipeline you can convert the mapped abundance statistics into a read abundance table that can be used as an abundance table in R. 

python2 get_count_table.py *.idxstats.txt > counts_baculo.txt
sed 's/.idxstats.txt//g' counts_baculo.txt > counts2_baculo.txt

#this was further edited in excel to remove all reference accession genomes that had zero mapped reads across all samples to make the file smaller in size (counts2_baculo_edited.csv)

###########################################
## ------------------------------------- ##
## ---- Baculoviridae Taxonomy File ---- ##
## ------------------------------------- ##
###########################################

#The Baculoviridae taxonomy file (baculo_taxa.csv) was generated by taking all the accession numbers associated with the reference genomes (baculoviridae.fasta) used and putting them in an excel file
#All accession numbers that did not have any mapped reads across all samples were removed from the file
#The remaining accession numbers (that had at least one mapped read in a sample) were annotated using their associated taxonomy information provided on NCBI
